internal/datasources/http/data_source_http.go

```diff
@@ -100,11 +114,25 @@ func (d *dataSource) Read(ctx context.Context, req tfsdk.ReadDataSourceRequest,
 	}
 
 	url := model.URL.Value
-	headers := model.RequestHeaders
+	method := model.Method.Value
+	requestHeaders := model.RequestHeaders
+	requestBody := strings.NewReader(model.RequestBody.Value)
+
+	if method == "" {
+		method = "GET"
+	}
+
+	if !isAllowedMethod(method) {
+		resp.Diagnostics.AddError(
+			"Method not allowed",
+			fmt.Sprintf("Method %q not allowed, must be one of: \"%s\".", method, strings.Join(allowedMethods, `", "`)),
+		)
+		return
+	}
```

Comment:
Drive-by note: This can use the shiny new terraform-plugin-framework-validators v0.3.0 functionality now 👍 

```go
Validators: []tfsdk.AttributeValidator{
    stringvalidator.OneOf([]string{"GET", "POST", "PUT", "PATCH", "DELETE", "HEAD"}...),
}
```

Reply:
Nice! Good point. Have refactored.